### PR TITLE
[2.5] cleanup deprecated uses

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

I wish deprecation authors could do this job.

### CODE REVIEW

File: src/Symfony/Component/HttpKernel/EventListener/ProfilerListener.php [hunk 1]
```diff
@@ -73,7 +73,7 @@ public function onKernelException(GetResponseForExceptionEvent $event)
     }
 
     /**
-     * @deprecated Deprecated since version 2.4, to be removed in 3.0.
+     * @deprecated Deprecated since version 2.4, to be removed in 3.0. REALLY???
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/EventListener/ProfilerListener.php]: Is this really deprecated?

Comment (Tobion) [inline src/Symfony/Component/HttpKernel/EventListener/ProfilerListener.php]: Yes because it is not needed anymore with the request stack.

Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/EventListener/ProfilerListener.php]: Understood, thanks

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

File: src/Symfony/Component/Validator/Tests/Constraints/AbstractConstraintValidatorTest.php [hunk 2]
```diff
@@ -53,6 +53,10 @@
 
     protected function setUp()
     {
+        if (Validation::API_VERSION_2_5 !== $this->getApiVersion()) {
```
Comment (stof) [inline src/Symfony/Component/Validator/Tests/Constraints/AbstractConstraintValidatorTest.php]: Technically, tests for the 2.5-BC API layer should not be allowed to trigger the warnings (the code should still rely on the 2.5 API when it has access to both APIs)

File: src/Symfony/Component/Validator/ValidatorBuilder.php [hunk 3]
```diff
@@ -377,18 +385,15 @@ public function getValidator()
                 $loader = $loaders[0];
             }
 
-            $metadataFactory = new ClassMetadataFactory($loader, $this->metadataCache);
+            if (Validation::API_VERSION_2_5 === $apiVersion) {
```
Comment (stof) [inline src/Symfony/Component/Validator/ValidatorBuilder.php]: this should be done for 2.5-BC as well actually

**Pull request merged**